src/sql/src/session/vars.rs

```diff
@@ -1483,6 +1483,14 @@ pub const ENABLE_COLUMNATION_LGALLOC: ServerVar<bool> = ServerVar {
     internal: true,
 };
 
+pub const ENABLE_EAGER_DELTA_JOINS: ServerVar<bool> = ServerVar {
+    name: UncasedStr::new("enable_eager_delta_joins"),
+    value: &false,
+    description:
+        "Plan delta joins when it would require no more arrangements than a differential join.",
+    internal: false,
```

Comment:
Yeah, I'd say let's make it internal for now.

Reply:
Later, if we want to make this switchable by users, then I think we'll probably want to expose this switch to users as a query hint instead of a system-wide flag. (A system-wide flag affecting the optimizer is a bit messy, because it applies only when you optimize a query, so the user has to pay attention to replan everything. If the user doesn't replan everything after switching the flag, then we have a risk of regressing some plans during the restart of a system upgrade, which can lead to confusing incidents: similar to incident-86, but even worse, because it would be hard to figure out why the plans changed during the upgrade.)

Reply:
@teskje: I think @ggevay is right - the workflow that I envision is:

1. Add a feature flag and plumb it through the `OptimizerConfig` and the config structs derived from that until the right point of the optimizer.
2. Run experiments and verify that the optimization makes a difference.
3. If there are only improvements and no regressions, turn it on globally. This will be quite rare.
4. If there are some improvements and some regressions, we'll need to make calculated decisions.
    1. Roll out to specific environments and bite the bullet for the known regressions.
    2. Rework the feature flag as an optimizer hint so we can teach specific queries to support it. 

I would like to be able to do (4.ii) without spending days in plumbing things through, though, and on top of syntax that is a bit more consistent than the current `OPTIONS` clause.

Reply:
This plan is good; I've moved over to `feature_flags!` for now.